crates/uv-pep508/src/lib.rs

```diff
@@ -144,40 +144,60 @@ impl<T: Pep508Url> Requirement<T> {
             self.version_or_url = None;
         }
     }
+
+    /// Returns string representation without masking credentials.
+    pub fn to_string_with_credentials(&self) -> String {
```

Comment:
We can give this a more native-feeling API with a proxy type (eliding generics):
```rust
struct RequirementDisplay<'a> {
    display_credentials: bool,
    requirement: &'a Requirement,
}

impl Display for RequirementDisplay<'_> {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        // inlining fmt_requirement
    }
}
```